.github/scripts/decomp-dev.py

```diff
@@ -0,0 +1,149 @@
+    total_functions = len(funcs)
+
+    matched_code = sum([fun["size"] for fun in funcs if fun["status"] == "Matching"])
+    matched_functions = len([fun for fun in funcs if fun["status"] == "Matching"])
+    matched_code_percent = matched_code / total_code * 100 if total_code > 0 else 0.0
+    matched_functions_percent = matched_functions / total_functions * 100 if total_functions > 0 else 0.0
+
+    fuzzy_code = sum([fun["size"] for fun in funcs if fun["status"] != "NotDecompiled"])
+    fuzzy_code_percent = fuzzy_code / total_code * 100 if total_code > 0 else 0.0
+    fuzzy_match_percent = fuzzy_code_percent * 0.5  # TODO: 0.5 for data being missing entirely
+
+    # TODO: we don't handle data or linking yet, so 0 for all of those
+    total_data = 0
+    matched_data = 0
+    matched_data_percent = 0.0
+    complete_code = 0
+    complete_code_percent = 0.0
+    complete_data = 0
+    complete_data_percent = 0.0
+    complete_units = 0
+    
+    measures = Measures()
+    measures.fuzzy_match_percent = fuzzy_match_percent
+    measures.total_code = total_code
+    measures.matched_code = matched_code
+    measures.matched_code_percent = matched_code_percent
+    measures.total_data = total_data
+    measures.matched_data = matched_data
+    measures.matched_data_percent = matched_data_percent
+    measures.total_functions = total_functions
+    measures.matched_functions = matched_functions
+    measures.matched_functions_percent = matched_functions_percent
+    measures.complete_code = complete_code
+    measures.complete_code_percent = complete_code_percent
+    measures.complete_data = complete_data
+    measures.complete_data_percent = complete_data_percent
+    measures.total_units = total_units
+    measures.complete_units = complete_units
+    return measures
+
+def get_functions(unit_name):
+    funcs = data[unit_name][".text"]
+    functions = []
+
+    # TODO: determine actual percentage based on tools/check?
+    def get_function_fuzzy_match_percent(fun):
+        if fun["status"] == "Matching":
+            return 100.0
+        elif fun["status"] == "NonMatchingMinor":
+            return 75.0
+        elif fun["status"] == "NonMatchingMajor":
+            return 25.0
+        else:
+            return 0.0
+        
+    def get_function_demangled_name(name):
+        try:
+            return cxxfilt.demangle(name)
+        except cxxfilt.InvalidName:
+            return name
+    
+    for fun in funcs:
+        function = ReportItem()
+        function.name = fun["label"] if isinstance(fun["label"], str) else fun["label"][0]
+        function.size = fun["size"]
+        function.fuzzy_match_percent = get_function_fuzzy_match_percent(fun)
+
+        metadata = ReportItemMetadata()
+        metadata.demangled_name = get_function_demangled_name(function.name)
+        metadata.virtual_address = fun["offset"]
+
+        function.metadata.CopyFrom(metadata)
+        function.address = fun["offset"]
```

Comment:
This should be relative to the beginning of the unit, (name is confusing, sorry) you can set it to null if that's too complicated